### PR TITLE
Task #6 Method setValueFrameAndPrevFrames() in class Game

### PR DESCRIPTION
**Pull request by Tomek52**

### CODE REVIEW

File: tests/GameTests.cpp [hunk 1]
```diff
@@ -95,3 +95,97 @@ TEST_F(GameTests, score_should_be_three)
...
+    //GIVEN
+    Game game;
+    //WHEN
+    game.setValueFrameAndPrevFrames(1, "X");
+    game.setValueFrameAndPrevFrames(2, "6/");
+    game.setValueFrameAndPrevFrames(3, "--");
+    //THEN
+    ASSERT_EQ(20, game.getFrame(1).getValue());
+    ASSERT_EQ(10, game.getFrame(2).getValue());
+}
```
Comment (Angan7a) [inline tests/GameTests.cpp]: The following tests will not pass:
```
TEST_F(GameTests, given_X_34)
{
    //GIVEN
    Game game;
    //WHEN
    game.setValueFrameAndPrevFrames(0, "X");
    game.setValueFrameAndPrevFrames(1, "34");
    //THEN
    ASSERT_EQ(17, game.getFrame(0).getValue());
    ASSERT_EQ(7, game.getFrame(1).getValue());
}

TEST_F(GameTests, given_34_4_no_full_frame)
{
    //GIVEN
    Game game;
    //WHEN
    game.setValueFrameAndPrevFrames(0, "34");
    game.setValueFrameAndPrevFrames(1, "4");
    //THEN
    ASSERT_EQ(7, game.getFrame(0).getValue());
    ASSERT_EQ(4, game.getFrame(1).getValue());
}
```

File: src/Game.cpp [hunk 2]
```diff
@@ -27,24 +28,62 @@ void Game::setIsStrikeOrSpare(unsigned short int position, const std::string& va
...
+std::pair <int, int> Game::convertValueToInt(const std::string& value)
+{
+    int valInt1 = 0;
+    int valInt2 = 0;
+
+    auto valueTemp = value.substr(0, 2);
+    std::replace(valueTemp.begin(), valueTemp.end(), '-', '0');
+
+    if (value[0] == 'X') valInt1 = 10;
+    else if (value[1] == '/')
```
Comment (Angan7a) [inline src/Game.cpp]: It should be check `else if (value.size() == 2 && value[1] == '/')`, because if will be no full frame can be problem

**Comment by anorberciak** [comment]
Could you only change the return value in checkValue(), from int to pair <int, int>. Maybe it will be better, instead of adding a new convertValueToInt() function.

File: src/Game.cpp [hunk 3]
```diff
@@ -27,24 +28,62 @@ void Game::setIsStrikeOrSpare(unsigned short int position, const std::string& va
...
 {
-    int valueInt;
-    auto valueTemp=value.substr(0,2);
-    if ((value[0] == 'X') || (value[1] == '/'))
+    int firstBall, secondBall;
+    std::tie(firstBall, secondBall) = convertValueToInt(value);
+    return firstBall + secondBall;
+}
+
+std::pair <int, int> Game::convertValueToInt(const std::string& value)
```
Comment (LordLukin) [inline src/Game.cpp]: The function name does not tell what this function is doing. 
convertValueToInts or convertValueToPairOfInts is more proper now.

File: tests/GameTests.cpp [hunk 4]
```diff
@@ -95,3 +95,97 @@ TEST_F(GameTests, score_should_be_three)
     ASSERT_EQ(3, checkValue);
 }
 
+TEST_F(GameTests, first_ball_should_be_3_and_second_4)
```
Comment (LordLukin) [inline tests/GameTests.cpp]: test names are not proper. Please change them to something like this: GivenValue34ConvertValuesToPairOfIntsShouldReturn3And4

File: tests/GameTests.cpp [hunk 5]
```diff
@@ -95,3 +95,97 @@ TEST_F(GameTests, score_should_be_three)
...
+    //THEN
+    ASSERT_EQ(3, firstBall);
+    ASSERT_EQ(4, secondBall);
+}
+
+TEST_F(GameTests, first_ball_should_be_10_and_second_0)
+{
+    //GIVEN
+    Game game;
+    int firstBall, secondBall;
```
Comment (LordLukin) [inline tests/GameTests.cpp]: Please extract common variables into GameTests class, so that you will not have to create them in every test.

**Pull request merged**